### PR TITLE
PIM-6847: Manage product and variant product history on frontend side

### PR DESCRIPTION
**Pull request by damien-carcel**
## Description

A first improvement was made in #6905 to allow variant product history to be displayed. This was done on the backend side, but was not an ideal solution (very dirty in fact).

This PR reverts the development previously made, and replaces it with a full frontend solution.

**Important** The first commit of this PR comes from my previous PR: #6929. I think I just forgot to push it... 😑 

## Definition Of Done

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Added integration tests           | -
| Changelog updated                 | OK
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: CHANGELOG-2.0.md [hunk 1]
```diff
@@ -5,6 +5,10 @@
 - PIM-6865: Fix ACL on import profile page
 - PIM-6876: Escape u001f character to workaround a mysql bug
 
+## Better manage products with variants!
+
+- PIM-6863: Hide "Variant" meta in non variant products
```
Comment (damien-carcel) [inline CHANGELOG-2.0.md]: Forgotten in a previous PR

File: features/product/display_variant_product.feature [hunk 2]
```diff
@@ -0,0 +1,16 @@
+@javascript
+Feature: Display a variant product
+  In order to enrich the catalog
+  As a regular user
+  I need to be able display a variant product with their specific informations
```
Comment (grena) [inline features/product/display_variant_product.feature]: Never "s" on **information** in english :)

> (...) with **its** specific (...)

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/history.js [hunk 3]
```diff
@@ -114,13 +116,28 @@ define(
...
+                    { entityId: entityMeta.id }
                 ).then(this.addAttributesLabelToVersions.bind(this));
             },
 
             /**
+             * @param {Object} entityMeta
+             *
+             * @returns Fetcher
+             */
+            getHistoryFetcher: function (entityMeta) {
```
Comment (grena) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/history.js]: Imho you should send directly the entity, it's weird to send only the meta

File: src/Pim/Bundle/VersioningBundle/Builder/VersionBuilder.php [hunk 4]
```diff
@@ -49,8 +44,7 @@ public function __construct(NormalizerInterface $normalizer, VersionFactory $ver
      */
     public function buildVersion($versionable, $author, Version $previousVersion = null, $context = null)
     {
-        $resourceName = $this->getResourceName($versionable);
-
+        $resourceName = ClassUtils::getClass($versionable);
```
Comment (grena) [inline src/Pim/Bundle/VersioningBundle/Builder/VersionBuilder.php]: Can't we use `get_class` now?

File: src/Pim/Bundle/VersioningBundle/Builder/VersionBuilder.php [hunk 5]
```diff
@@ -84,10 +75,8 @@ public function buildVersion($versionable, $author, Version $previousVersion = n
      */
     public function createPendingVersion($versionable, $author, array $changeset, $context = null)
     {
-        $resourceName = $this->getResourceName($versionable);
-
         $version = $this->versionFactory->create(
-            $resourceName,
+            ClassUtils::getClass($versionable),
```
Comment (grena) [inline src/Pim/Bundle/VersioningBundle/Builder/VersionBuilder.php]: same question

File: src/Pim/Bundle/VersioningBundle/Builder/VersionBuilder.php (hunk 4 above)
Comment (damien-carcel) [reply src/Pim/Bundle/VersioningBundle/Builder/VersionBuilder.php]: It is a revert :)

File: src/Pim/Bundle/VersioningBundle/Builder/VersionBuilder.php (hunk 5 above)
Comment (damien-carcel) [reply src/Pim/Bundle/VersioningBundle/Builder/VersionBuilder.php]: Same answer ;)

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/history.js (hunk 3 above)
Comment (damien-carcel) [reply src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/history.js]: Will disappear in coming PR on product model history.

**Pull request merged**